### PR TITLE
[sdks] Add provisioning for Android SDK and NDK

### PR DESCRIPTION
**Pull request by luhenry**
This will also be used by XA

### CODE REVIEW

**Comment by luhenry** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by luhenry** [comment]
@monojenkons build OS X x64 Android SDK

**Comment by akoeplinger** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by akoeplinger** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by luhenry** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by luhenry** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by luhenry** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by luhenry** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by luhenry** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by luhenry** [comment]
@monojenkins build OS X x64 Android SDK

File: sdks/builds/mxe.mk [hunk 1]
```diff
@@ -1,31 +1,19 @@
 
 MXE_SRC?=$(TOP)/sdks/builds/toolchains/mxe
-MXE_PREFIX?=$(TOP)/sdks/out/mxe
+MXE_PREFIX_DIR?=$(TOP)/sdks/out
 
-$(TOP)/sdks/builds/toolchains/mxe:
-	git clone -b xamarin https://github.com/xamarin/mxe.git $@
-	cd $@ && git checkout $(MXE_HASH)
+# This is not overridable
+MXE_PREFIX=$(MXE_PREFIX_DIR)/mxe-$(shell echo $(MXE_HASH) | head -c 6)
```
Comment (akoeplinger) [inline sdks/builds/mxe.mk]: I don't think we really need to shorten MXE_HASH here, the full sha is OK.

At the very least we should be using `:=` for variable assignment so we don't shell out each time this variable is used.

**Comment by luhenry** [comment]
@monojenkins bacport to 2017-12

**Comment by luhenry** [comment]
@monojenkins backport to 2017-12

**Comment by luhenry** [comment]
@monojenkins backport to 2018-02

**Comment by luhenry** [comment]
@monojenkins backport to 2018-04

File: sdks/builds/mxe.mk (hunk 1 above)
Comment (luhenry) [reply sdks/builds/mxe.mk]: Shortening `MXE_HASH` is the approach taken by XA, so for the sake of ease of integration, I didn't want to do it differently.

**Pull request merged**